agent/test-agent-cli/design/RUNBOOK.md

```diff
@@ -0,0 +1,82 @@
+# Steps to use Test-agent Command Line Interface
+`test-agent-cli` lets you write Testsys test using Bash and help in receiving and sending information from/to the Testsys cluster.
+
+## Prerequisites:
+* kind: https://kind.sigs.k8s.io/docs/user/quick-start/
+* Kubectl: https://kubernetes.io/docs/tasks/tools/
+* Docker : https://docs.docker.com/get-started/
+
+## Steps to install Testsys
+1. Set the TESTSYS_DIR variable to point to the directory in which you have cloned the project. For example:`export TESTSYS_DIR="${HOME}/repos/bottlerocket-test-system"`
+2. Set alias `alias testsys="${TESTSYS_DIR}/.cargo/bin/testsys"`
+3. Install the testsys command line tool into the local CARGO_HOME as: 
+```shell
+cd "${TESTSYS_DIR}"
+cargo install --path "${TESTSYS_DIR}/bottlerocket/testsys" --force
+```
+
+## Steps to create Bash based TestAgent:
+
+1. Create a Bash Test using the following commands to achieve the desired behavior of test.
+```shell
+# To get the configuration details and setting the task state running
+test-agent-cli init
+
+# To get the number of retires allowed in case of failing tests
+test-agent-cli retry-count
+
+# To send the result of every test run to test object in Controller
+test-agent-cli send-result -o pass -p 1 -f 0 -s 0
+
+# To get the secret value using the secret key
+test-agent-cli get-secret secret-key
+
+# To send any error encountered in test
+test-agent-cli send-error error-message
+
+# To create the Tar file of the results, Marking task_state as completed and handling keep_running behaviour
+# Send the path and name of results directory where all the results have been saved
+test-agent-cli terminate --results-dir results_directory
+```
+2. Create a docker file(to provide all the required resources to run the Bash script along with the Command Line Interface in a single pod), that will be used to create image that will run on the Test Pod. Example Dockerfile : [Example Agent Dockerfile](../examples/example_test_agent_cli/Dockerfile).\
+   In this Dockerfile make sure to set the `ENTRYPOINT` to the test Bash script created in Step 1. Add the commands to install the packages required for Bash script in the Dockerfile.
+3. Move to [root](https://github.com/bottlerocket-os/bottlerocket-test-system) of the repository.
+4. Create a docker image consists of the Command Line Interface, Bash Script test and all other resources required to run the test: 
```

Comment:
4. Create the docker image.
script (image can be tagged here in the build command)